include/behaviortree_cpp/basic_types.h

```diff
@@ -450,6 +452,43 @@ struct has_static_method_description<
 {
 };
 
+/// Optional metadata for a TreeNodeManifest.
+/// The metadata can represent an xml element with text:
+///   <metadata_name>Descriptive text</metadata_name>
+/// Or an xml element with an attribute:
+///   <metadata_name attribute="value"/>
+struct ManifestMetadata
+{
+  /// A pair containing the <name, value> of a metadata's XML attribute.
+  using Attribute = std::pair<std::string, std::string>;
+
+  [[nodiscard]] bool representsText() const
+  {
+    return std::holds_alternative<std::string>(text_or_attribute);
+  }
+
+  [[nodiscard]] bool representsAttribute() const
+  {
+    return std::holds_alternative<Attribute>(text_or_attribute);
+  }
+
+  std::string name;
+  std::variant<std::string, Attribute> text_or_attribute;
+};
```

Comment:
Too much complexity for a niche feature, in my opinion.
I would keep it simple and support only the attribute case.
Therefore:
```cpp
struct Metadata {
  std::string name;
  std::string value;
};
```